source/common/http/conn_manager_impl.cc

```diff
@@ -607,12 +607,13 @@ void ConnectionManagerImpl::ActiveStream::traceRequest() {
   ConnectionManagerImpl::chargeTracingStats(tracing_decision.reason,
                                             connection_manager_.config_.tracingStats());
 
-  if (!tracing_decision.is_tracing) {
+  active_span_ = connection_manager_.tracer_.startSpan(*this, *request_headers_, request_info_,
+                                                       tracing_decision);
+
+  if (!active_span_) {
```

Comment:
I can't remember now, but when does this return nullptr? I thought we always have a tracer and this will always return a span (even if the null span).

Reply:
No, current behaviour at start of `traceRequest()` is to return if not being sampled, before the span is created.

Reply:
Ah ok. Got it. Thanks.